opcua_tools/nodeset_generator.py

```diff
@@ -456,3 +462,31 @@ def find_namespaces_in_use(nodes, references, namespace_index):
     namespaces_in_use = list(set(namespaces_in_use + browsename_namespaces))
 
     return namespaces_in_use
+
+
+def create_required_models(
+    models: List[ua_models.UAModel],
+    model_uri: str,
+) -> str:
+    required_models_str = ""
+    try:
+        model = next(model for model in models if model.model_uri == model_uri)
+    except StopIteration:
+        return required_models_str
+
+    for required_model in model.required_models:
+        required_model_to_add = """\n        <RequiredModel ModelUri="{}" Version="{}" PublicationDate="{}" />""".format(
+            required_model.model_uri,
+            required_model.version,
+            required_model.publication_date
+            if required_model.publication_date is not None
+            else datetime.now(tz=pytz.UTC),
+        )
+        required_models_str = "{}{}".format(
+            required_models_str,

```

Comment:
Out of curiosity, is there a reason that you prefer this format rather than using fstrings?

Reply:
For this particular example, I went back to have a look at some performance charts:

https://stackoverflow.com/questions/1316887/what-is-the-most-efficient-string-concatenation-method-in-python

Using `.format()` is faster for concatenation than f-strings.
It doesn't do much for just concatenating 2 variables together, but it's faster and, in my opinion, readable at a similar level.